### PR TITLE
Add patterns for AWS CodeCommit as a CDK pipeline repository and as a ArgoCD workloads repository.

### PR DESCRIPTION
**Pull request by ybezsonov**
*Description of changes:*

1. A pattern shows usage of AWS CodeCommit as a source repository for CDK pipeline. 
- Multiple clusters with deployment pipeline with repository in CodeCommmit.
- Deployment instructions are in /lib/pipeline-codecommit-stack/index.ts

2. A pattern shows usage of AWS CodeCommit as a source repository for ArgoCD workloads.
- Single cluster, ArgoCD GitOps bootstrapped and exposed via Elastic Load Balancer
- AWS CodeCommmit as a repository for ArgoCD workloads, AWS Lambda which automatically triggers ArgoCD App of Apps resync after commit into AWS CodeCommit workloads repository
- Stack deploy (run script from repository root folder): ./lib/workloads-codecommit-construct/deploy.sh
- Stack destroy ./lib/workloads-codecommit-construct/destroy.sh

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by shapirov103 — COMMENTED** [review]
@ybezsonov apologies for the delay. I left a few comments, however most significant is adding documentation describing how to use the pattern and what the solution.
We may need to clean up a few things. I will be upgrading the patterns to blueprints 1.5.3 and can validate this once I understand the mechanics. 
Thank you for your contribution!

File: lib/workloads-codecommit-repo-stack/index.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  {
+    "ref": "refs/heads/master",
+    "repository": {
+      "html_url": "https://git-codecommit.us-west-2.amazonaws.com/v1/repos/eks-blueprints-workloads-cc",
+      "default_branch": "master"
+    }
+  }
+*/
+
+exports.handler = async function(event) {
```
[the remark this answers is not in the document]
Comment (ybezsonov) [reply lib/workloads-codecommit-repo-stack/index.ts]: I moved lamda code to ./lib/workloads-codecommit-construct/lambda/index.js

File: lib/workloads-codecommit-construct/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
+echo Deploying blueprint ...
```
[the remark this answers is not in the document]
Comment (ybezsonov) [reply lib/workloads-codecommit-construct/deploy.sh:1]: ./lib/workloads-codecommit-construct/deploy.sh is helper scrips which is example how the pattern can be used. It creates EKS cluster using blueprint, creates CodeCommit user, stack and automation, clones workloads example repository, copy and push it to CodeCommit stack so developer can work with it.
./lib/workloads-codecommit-construct/destroy.sh cleans all up.

Those scripts can be executed from the root folder of the pattern repository, for example in Cloud9.

Both those scripts can be used as documentation in addition to comments in ./bin/main.ts

File: lib/workloads-codecommit-repo-stack/index.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import { Construct } from 'constructs';
+import * as codecommit from 'aws-cdk-lib/aws-codecommit';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+
+export default class WorkloadsCodeCommitRepoStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, repoName: string, userName: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    const argoCDUrl = new cdk.CfnParameter(this, 'argoCDUrl', {
```
[the remark this answers is not in the document]
Comment (ybezsonov) [reply lib/workloads-codecommit-repo-stack/index.ts:11]: ArgoCD uses webhook to start Sync of the workloads repository when push to the repository heppens. 
When developer push code to CodeCommit repository, push event triggers Lamda which sends payload to argoCDUrl defined by this parameter. argoCDUrl is url of ArgoCD UI which is exposed by ELB during the blueprint bootstrapping.
values: { server: {service: { type: "LoadBalancer"} } }

**Comment by elamaran11** [comment]
@ybezsonov Are you still working on this PR. No activity for two months.

**Comment by elamaran11** [comment]
@ybezsonov Are you still working on this PR. No activity for 5 months.

**Comment by ybezsonov** [comment]
> @ybezsonov Are you still working on this PR. No activity for 5 months.
@elamaran11 

I will refresh it with the latest blueprint version in January and will push.

**Pull request closed without merging**